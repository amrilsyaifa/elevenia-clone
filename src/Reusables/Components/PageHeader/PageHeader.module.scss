@import 'src/Reusables/Styles/Variables';

.container {
  display: flex;
  flex-direction: column;
  width: 100%;
}
.wrapper-right-text {
  font-size: 14px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  color: $grey;
  margin-top: 10px;
  .wrapper-text {
    width: 150px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    .text {
      cursor: pointer;
    }
  }
}

.header-wrapper {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 70px;
  justify-content: space-around;
  img {
    width: 100px;
    height: 50px;
    cursor: pointer;
  }
  .search {
    margin-top: 15px;
    margin-left: 30px;
  }
}

.header-wrapper-single {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 70px;
  img {
    width: 100px;
    height: 50px;
    cursor: pointer;
  }
  .search {
    margin-top: 15px;
    margin-left: 30px;
  }
}
