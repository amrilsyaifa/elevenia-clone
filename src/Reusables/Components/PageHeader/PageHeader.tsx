import { Typography, Input } from 'antd';
import React from 'react';
import Basket from '../Basket';
import { useAppSelector } from 'src/Hooks/useReduxHooks';
import Style from './PageHeader.module.scss';

const { Text } = Typography;
const { Search } = Input;

const PageHeader: React.FC = () => {
  const onSearch = (value: string) => console.log(value);
  const data = useAppSelector((state) => state.checkout.data);
  return (
    <div className={Style['container']}>
      <div className={Style['wrapper-right-text']}>
        <div className={Style['wrapper-text']}>
          <Text className={Style['text']}>Login</Text>
          <Text className={Style['text']}>Logout</Text>
          <Basket total={data.length} />
        </div>
      </div>
      <div className={Style['header-wrapper']}>
        <img src="/assets/images/logo.png" alt="logo" />
        <Search
          className={Style['search']}
          placeholder="Cari Produk"
          allowClear
          enterButton="Cari"
          size="large"
          onSearch={onSearch}
        />
      </div>
    </div>
  );
};

export default PageHeader;
