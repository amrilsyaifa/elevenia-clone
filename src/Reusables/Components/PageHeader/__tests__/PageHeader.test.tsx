import { render, screen } from '@testing-library/react';
import React from 'react';
import PageHeader from '../PageHeader';

describe('PageHeader', () => {
  test('PageHeader Render Snapshoot', () => {
    const { container } = render(<PageHeader />);
    expect(container).toMatchSnapshot();
  });

  test('PageHeader Have Title', () => {
    render(<PageHeader />);
    const titleHeader = screen.getByText('Title');
    expect(titleHeader).toBeInTheDocument();
  });
});
