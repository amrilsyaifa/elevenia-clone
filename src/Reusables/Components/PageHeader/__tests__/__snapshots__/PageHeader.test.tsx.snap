// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader PageHeader Render Snapshoot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="wrapper-right-text"
    >
      <div
        class="wrapper-text"
      >
        <span
          class="ant-typography text"
        >
          Login
        </span>
        <span
          class="ant-typography text"
        >
          Logout
        </span>
        <div
          class="container"
          data-testid="basket"
        >
          <div
            class="wrapper-number"
          >
            <span
              class="ant-typography number"
            >
              0
            </span>
          </div>
          <span
            aria-label="shopping-cart"
            class="anticon anticon-shopping-cart shopping-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="shopping-cart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 00-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 100 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 00-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 00-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="header-wrapper"
    >
      <img
        alt="logo"
        src="/assets/images/logo.png"
      />
      <span
        class="ant-input-group-wrapper ant-input-search ant-input-search-large ant-input-search-with-button search ant-input-group-wrapper-lg"
      >
        <span
          class="ant-input-wrapper ant-input-group"
        >
          <span
            class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          >
            <input
              class="ant-input ant-input-lg"
              placeholder="Cari Produk"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                class="ant-input-clear-icon ant-input-clear-icon-hidden"
                role="button"
                tabindex="-1"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </span>
          <span
            class="ant-input-group-addon"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
              type="button"
            >
              <span>
                Cari
              </span>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
